### PR TITLE
Fix crm_diff --no-version

### PR DESCRIPTION
**Pull request by kgaillot**
with other minor fixes

### CODE REVIEW

**Comment by kgaillot** [comment]
@gao-yan Since you added crm_diff --no-version, you might be particularly interested in the second commit (20a74b9).

File: tools/xml_diff.c [hunk 1]
```diff
@@ -158,6 +158,13 @@ main(int argc, char **argv)
         crm_help('?', EX_USAGE);
     }
 
+    if (apply && no_version) {
+        fprintf(stderr, "warning: -u/--no-version ignored with -p/--patch\n");
+    } else if (as_cib && no_version) {
+        fprintf(stderr, "error: -u/--no-version incompatible with -c/--cib\n");
```
Comment (gao-yan) [inline tools/xml_diff.c]: AFAICS,  --cib in this case  basically determines to  add a digest in the diff. It doesn't have to be incompatible with --no-version.

File: tools/xml_diff.c [hunk 2]
```diff
@@ -199,6 +206,21 @@ main(int argc, char **argv)
...
+
+        /* If we're ignoring the version, make the version information
+         * identical, so it isn't detected as a change. */
+        if (no_version) {
+            for (lpc = 0; lpc < DIMOF(vfields); lpc++) {
+                crm_copy_xml_element(object_1, object_2, vfields[lpc]);
+            }
+        }
+
         xml_track_changes(object_2, NULL, object_2, FALSE);
```
Comment (gao-yan) [inline tools/xml_diff.c]: Good point and makes sense. Thanks!

File: tools/xml_diff.c (hunk 1 above)
Comment (kgaillot) [inline tools/xml_diff.c]: I'm not sure. Would a digest be any use if you're stripping version information?

Comment (gao-yan) [inline tools/xml_diff.c]: Hmm, probably not. Just recalled the fact that version information is taken into account when calculating digest.

**Pull request merged**